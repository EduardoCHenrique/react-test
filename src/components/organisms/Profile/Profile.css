@import '../../../style/index.css';

.profile {
  display: flex;
  align-items: center;
  height: var(--profile-height);
}

.profile__data {}

.profile__name {
  font-size: 3.2rem;
  font-weight: 400;
  margin: 0 0 2rem;
}

.profile__avatar {
  width: 10rem;
  height: 10rem;
  margin-right: 1rem;
}

.profile__bio {
  font-size: 1.4rem;
  font-weight: 400;
  color: var(--secondary-text-color);
  margin: 0 0 1rem 0;
}

.profile__details {
  font-size: 1.2rem;
  font-weight: 400;
  color: var(--secondary-text-color);
}

.profile__detail {
  margin-right: 2rem;
}

.profile__detail__icon {
  margin-right: .4rem;
}

.profile__detail__link:hover {
  color: inherit;
  text-decoration: underline;
}