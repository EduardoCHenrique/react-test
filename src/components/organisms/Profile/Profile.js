import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Avatar from 'src/components/atoms/Avatar';
import { GoLocation, GoLink } from "react-icons/go";

import './Profile.css';

const Profile = ({avatar_url, name, bio, location, blog, className}) => {
  return(
    <div className={classNames('profile', className)}>
      <Avatar className="profile__avatar" image={avatar_url} size='medium' />
      <div className="profile__data">
        <h2 className="profile__name">{name}</h2>
        <h3 className="profile__bio">{bio}</h3>
        <div className="profile__details">
          <span className="profile__detail">
            <GoLocation className='profile__detail__icon'/>
            {location}
          </span>
          <span className="profile__detail">
            <GoLink className='profile__detail__icon'/>
            <a href={blog} className="profile__detail__link" target="_blank">{blog}</a>
          </span>
        </div>
      </div>
    </div>
  )
}

Profile.propTypes = {
  avatar_url: PropTypes.string,
  name: PropTypes.string,
  bio: PropTypes.string,
  location: PropTypes.string,
  blog: PropTypes.string,
  className: PropTypes.string
}

export default Profile