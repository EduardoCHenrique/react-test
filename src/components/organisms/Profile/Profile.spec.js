import React from 'react';
import { shallow } from 'enzyme';
import Profile from './Profile.js';

const default_props = {};

function setup(props = default_props) {
  const wrapper = shallow(<Profile />);
  return { wrapper, props };
}

describe('Profile Test Suite', () => {
  it('Should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});