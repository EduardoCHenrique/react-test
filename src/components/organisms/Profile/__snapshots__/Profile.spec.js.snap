// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Test Suite Should match snapshot 1`] = `ShallowWrapper {}`;
